ega-data-api-htsget/src/main/java/eu/elixir/ega/ebi/htsget/service/internal/TicketServiceV2Impl.java

```diff
@@ -0,0 +1,203 @@
+import eu.elixir.ega.ebi.htsget.formats.DataProvider;
+import eu.elixir.ega.ebi.htsget.formats.DataProviderFactory;
+import eu.elixir.ega.ebi.htsget.service.TicketServiceV2;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.springframework.http.HttpRange;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+public class TicketServiceV2Impl implements TicketServiceV2 {
+
+    public static final int MAX_BYTES_PER_DATA_BLOCK = 1024 * 1024 * 1024;
+
+    private final FileInfoService fileInfoService;
+
+    private final MyExternalConfig externalConfig;
+
+    private final ResClient resClient;
+
+    private final DataProviderFactory dataProviderFactory;
+
+
+    public TicketServiceV2Impl(FileInfoService fileInfoService, MyExternalConfig externalConfig, ResClient resClient, DataProviderFactory dataProviderFactory) {
+        this.fileInfoService = fileInfoService;
+        this.externalConfig = externalConfig;
+        this.resClient = resClient;
+        this.dataProviderFactory = dataProviderFactory;
+    }
+
+    public HtsgetResponseV2 getFile(String id,
+                                    String format,
+                                    Optional<String> requestClass,
+                                    Optional<String> referenceName,
+                                    Optional<Long> start,
+                                    Optional<Long> end,
+                                    Optional<List<Field>> fields,
+                                    Optional<List<String>> tags,
+                                    Optional<List<String>> notags)
+            throws HtsgetException, NotFoundException, PermissionDeniedException, IOException, URISyntaxException {
+
+        boolean onlyHeader = false;
+        if (requestClass.isPresent()) {
+            if (!requestClass.get().equalsIgnoreCase("Header"))
+                throw new InvalidInputException("Invalid class parameter");
+            onlyHeader = true;
+            if (referenceName.isPresent() ||
+                    start.isPresent() ||
+                    end.isPresent() ||
+                    fields.isPresent() ||
+                    tags.isPresent() ||
+                    notags.isPresent())
+                throw new InvalidInputException("Other parameters not allow for Header request");
+        }
+
+        if (start.isPresent()) {
+            if (!referenceName.isPresent())
+                throw new InvalidInputException("Must specify reference name if using start parameter");
+
+            if (referenceName.get().equals("*"))
+                throw new InvalidInputException("Cannot use start with unplaced unmapped reads");
+        }
```

Comment:
Substitute for a an "assert" method call to improve readability

Reply:
I don't understand, do you mean I should extract this to a separate method?

Reply:
Yes, extract to a separate method asser[condition] that would throw the exception.

Reply:
I refactored all of the headerOnly checks out into one method. Do you think this is OK?